libcanard/canard.h

```diff
@@ -1,528 +1,587 @@
-/*
- * Copyright (c) 2016-2020 UAVCAN Development Team
- *
- * Permission is hereby granted, free of charge, to any person obtaining a copy
- * of this software and associated documentation files (the "Software"), to deal
- * in the Software without restriction, including without limitation the rights
- * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
- * copies of the Software, and to permit persons to whom the Software is
- * furnished to do so, subject to the following conditions:
- *
- * The above copyright notice and this permission notice shall be included in all
- * copies or substantial portions of the Software.
- *
- * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
- * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
- * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
- * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
- * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
- * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
- * SOFTWARE.
- *
- * Contributors: https://github.com/UAVCAN/libcanard/contributors
- */
-
-#ifndef CANARD_H
-#define CANARD_H
+///          LIBCANARD
+///
+/// Libcanard is a compact implementation of the UAVCAN/CAN protocol for high-integrity real-time embedded systems.
+/// It is designed for use in robust deterministic embedded systems equipped with at least 32K ROM and 4K RAM.
```

Comment:
I suspect that CAN FD may actually lead to a somewhat *lower* memory utilization because there will be fewer instances of per-frame overhead. But anyway, I think a detailed resource utilization model would be out of place in this high-level overview of the library. I replaced "4K" with "4..8K" to make it more explicit that this is not meant to be a hard design requirement but rather a handwavy estimate. Users who need a more detailed understanding of how much memory is needed will find the necessary information in the description that follows.

Reply:
If you don't think FD uses significantly more memory then I'm fine with the comment.